src/main/app/address/AddressLookupPostControllerBase.ts

```diff
@@ -0,0 +1,75 @@
+import autobind from 'autobind-decorator';
+import { Response } from 'express';
+
+import { FieldPrefix } from '../../app/case/case';
+import { AppRequest } from '../controller/AppRequest';
+import { AnyObject, PostController } from '../controller/PostController';
+import { Form, FormFields, FormFieldsFn } from '../form/Form';
+import { Address, getAddressesFromPostcode } from '../postcode/postcode-lookup-api';
+
+@autobind
+export default class AddressLookupPostControllerBase extends PostController<AnyObject> {
+  constructor(protected readonly fields: FormFields | FormFieldsFn, protected readonly fieldPrefix: FieldPrefix) {
+    super(fields);
+  }
+
+  public async post(req: AppRequest<AnyObject>, res: Response): Promise<void> {
+    const postcode = req.body[`${this.fieldPrefix}AddressPostcode`] as string;
+
+    let addresses;
+
+    const fields = typeof this.fields === 'function' ? this.fields(req.session.userCase) : this.fields;
+    const form = new Form(fields);
+    const { saveAndSignOut, saveBeforeSessionTimeout, _csrf, ...formData } = form.getParsedBody(req.body);
+
+    req.session.errors = form.getErrors(formData);
+
+    Object.assign(req.session.userCase, formData);
+
+    if (req.session.errors.length === 0) {
+      const stubbedPostcode = this.checkStubbedPostcode(postcode);
+      if (stubbedPostcode) {
+        addresses = stubbedPostcode;
+      } else {
+        addresses = await getAddressesFromPostcode(postcode, req.locals.logger);
+      }
+      req.session.addresses = addresses;
+    }
+
+    this.redirect(req, res);
+  }
+
+  private checkStubbedPostcode(postcode: string): Address[] | null {
```

Comment:
I believe this stub method was added for testing screen changes until we didn't had the actual api key. If thats the case, can we please remove this stub method and its respective calls?

Reply:
Done - removed the method and all uses.